--- pisi/constants.py~	2013-02-14 18:56:41.014485399 +0100
+++ pisi/constants.py	2013-02-14 19:13:40.331448155 +0100
@@ -97,6 +97,10 @@
         self.__c.needs_reboot = "needsreboot"
         self.__c.files_db = "files.db"
         self.__c.repos = "repos"
+        self.__c.devel_package_end = "-devel"
+        self.__c.assign_to_system_devel = ["system.base", "system.devel"]
+        self.__c.system_devel_component = "system.devel"
+        self.__c.devels_component = "programming.devel"
 
         #file/directory permissions
         self.__c.umask = 0022
--- pisi/operations/build.py~	2013-02-14 18:48:42.384502887 +0100
+++ pisi/operations/build.py	2013-02-14 19:15:52.196443337 +0100
@@ -1081,7 +1081,13 @@
             if not package.description:
                 package.description = self.spec.source.description
             if not package.partOf:
-                package.partOf = self.spec.source.partOf
+                if not package.name.endswith(ctx.const.devel_package_end):
+                    package.partOf = self.spec.source.partOf
+                else:
+                    if self.spec.source.partOf in ctx.const.assign_to_system_devel:
+                        package.partOf = ctx.const.system_devel_component
+                    else:
+                        package.partOf = ctx.const.devels_component
             if not package.license:
                 package.license = self.spec.source.license
             if not package.icon:
